arbnode/node.go

```diff
@@ -1593,3 +1587,75 @@ func (n *Node) ExpectChosenSequencer() containers.PromiseInterface[struct{}] {
 func (n *Node) BlockMetadataAtMessageIndex(msgIdx arbutil.MessageIndex) containers.PromiseInterface[common.BlockMetadata] {
 	return containers.NewReadyPromise(n.TxStreamer.BlockMetadataAtMessageIndex(msgIdx))
 }
+
+func (n *Node) GetL1Confirmations(msgIdx arbutil.MessageIndex) containers.PromiseInterface[uint64] {
+	// batches not yet posted have 0 confirmations but no error
+	batchNum, found, err := n.InboxTracker.FindInboxBatchContainingMessage(msgIdx)
+	if err != nil {
+		return containers.NewReadyPromise(uint64(0), err)
+	}
+	if !found {
+		return containers.NewReadyPromise(uint64(0), nil)
+	}
+	parentChainBlockNum, err := n.InboxTracker.GetBatchParentChainBlock(batchNum)
+	if err != nil {
+		return containers.NewReadyPromise(uint64(0), err)
+	}
+
+	if n.L1Reader == nil {
+		return containers.NewReadyPromise(uint64(0), nil)
+	}
+	if n.L1Reader.IsParentChainArbitrum() {
+		return stopwaiter.LaunchPromiseThread(n.L1Reader, func(ctx context.Context) (uint64, error) {
+			parentChainClient := n.L1Reader.Client()
+			parentChainBlock, err := parentChainClient.BlockByNumber(ctx, new(big.Int).SetUint64(parentChainBlockNum))
+			if err != nil {
+				// Hide the parent chain RPC error from the client in case it contains sensitive information.
+				// Likely though, this error is just "not found" because the block got reorg'd.
+				return 0, fmt.Errorf("failed to get parent chain block %v containing batch", parentChainBlockNum)
+			}
+
+			var confs uint64
+			err = parentChainClient.Client().CallContext(ctx, &confs, "arb_getL1Confirmations", parentChainBlock.Number())
+			if err != nil {
+				// falls back to node interface method
+				log.Debug("Failed to get L1 confirmations from parent chain via arb_getL1Confirmations", "blockNumber", parentChainBlockNum, "blockHash", parentChainBlock.Hash(), "err", err)
```

Comment:
I think we want to have a test covering that case (so L2 will not expose arb_getL1Confirmations.. possibly it can just not expose arb_ at all)

Reply:
Done.